cmd/podman/images/push.go

```diff
@@ -147,5 +151,14 @@ func imagePush(cmd *cobra.Command, args []string) error {
 
 	// Let's do all the remaining Yoga in the API to prevent us from scattering
 	// logic across (too) many parts of the code.
-	return registry.ImageEngine().Push(registry.GetContext(), source, destination, pushOptions.ImagePushOptions)
+	err := registry.ImageEngine().Push(registry.GetContext(), source, destination, pushOptions.ImagePushOptions)
+	if err != nil {
+		if errors.Cause(err) != storage.ErrImageUnknown {
```

Comment:
Having the error check here is unlikely to work in the remote client since we lose the type information over the wire.

I suggest something different and handle everything in `ImageEngine.Push` where we could do something as follows:
```golang
image := lookupLocalImage(...)
if isManifestList(image) && options.All {
    // Push manifest list and all images instead of just one image.
}
```

Reply:
You got me excited that an isManifestList object existed.

Reply:
Ah, sorry!  There should be code checking for the manifest types though.

Reply:
I actually rewrote a lot of this to be server side, so it should have better support for errors.Cause.